src/modules/45-dashboards/components/AidaDrawer/index.tsx

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2023 Harness Inc. All rights reserved.
+ * Use of this source code is governed by the PolyForm Shield 1.0.0 license
+ * that can be found in the licenses directory at the root of this repository, also available at
+ * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
+ */
+
+import React from 'react'
+import { Drawer, PopoverInteractionKind, PopoverPosition, Position } from '@blueprintjs/core'
+import { Color, FontVariation } from '@harness/design-system'
+import { Container, Icon, Layout, Popover, Text } from '@harness/uicore'
+import { useStrings } from 'framework/strings'
+import css from './AidaDrawer.module.scss'
+
+export interface AidaDrawerProps {
+  isOpen: boolean
+  setIsOpen: (isOpen: boolean) => void
+  children: React.ReactNode
+}
+
+const AidaDrawer: React.FC<AidaDrawerProps> = ({ children, isOpen, setIsOpen }) => {
+  const { getString } = useStrings()
+
+  const handleClose = (): void => {
+    setIsOpen(false)
+  }
+
+  return (
+    <>
+      <Drawer
+        position={Position.RIGHT}
+        isOpen={isOpen}
+        onClose={handleClose}
+        className={css.drawer}
+        isCloseButtonShown={true}
```

Comment:
Typically you don't need to pass `true`
```suggestion
        isCloseButtonShown
```